demo/test-extensions/Makefile

```diff
@@ -0,0 +1,18 @@
+CFLAGS ?= -Os -Wall
+
+PKG_MODULES   := gobject-2.0 webkit2gtk-web-extension-4.0
+WEB_EXT_FLAGS := $(shell pkg-config ${PKG_MODULES} --cflags)
+WEB_EXT_LIBS  := $(shell pkg-config ${PKG_MODULES} --libs)
+
+CPPFLAGS += ${WEB_EXT_FLAGS}
+LDLIBS   += ${WEB_EXT_LIBS}
```

Comment:
Looks like the WEB_EXT_* variables are one level of indirection for nothing, no?
I suggest setting CPPFLAGS and LDLIBS directly.

By the way, shouldn't it be CFLAGS instead of CPPFLAGS?  The latter is for the pre-processor, `cpp`.  If you want to set C++ flags, the right name is CXXFLAGS.